rules/python_rules.build_defs

```diff
@@ -572,12 +572,33 @@ def python_wheel(name:str, version:str, hashes:list=None, package_name:str=None,
 def _patch_cmd(patch):
     """Returns a command for the given patch or patches, with OS-specific tweaks where needed."""
     patches = [patch] if isinstance(patch, str) else patch
+
+    patch_opts = '' if CONFIG.HOSTOS == 'freebsd' else '--no-backup-if-mismatch'
+
+    cmds = []
+    for patch in patches:
+        # If the patch header contains the magic Git date, this is a patch
+        # generated with `git format-patch`, and needs to be applied with
+        # `patch -p1`; otherwise, default to `patch -p0`
+        cmds += [
+            "if sed -n 1p $(location {patch}) | grep -q '^From [0-9a-f]\{{40\}} Mon Sep 17 00:00:00 2001$'; then "
```

Comment:
Why'd this date hard coded? 

```
jpoole [gaia] ~/xattr (add-supported-const) ♬  git format-patch master
0001-Added-constant-for-whether-xattrs-are-supported-by-t.patch
jpoole [gaia] ~/xattr (add-supported-const) ♬  cat 0001-Added-constant-for-whether-xattrs-are-supported-by-t.patch
From fce5c00c88adef11e5f39505018e47aab1c6fb12 Mon Sep 17 00:00:00 2001
From: jpoole <jpoole@thoughtmachine.net>
Date: Mon, 5 Oct 2020 13:21:15 +0100
Subject: [PATCH] Added constant for whether xattrs are supported by the
 current platform

---
 xattr_bsd.go         | 3 +++
 xattr_darwin.go      | 3 +++
 xattr_linux.go       | 3 +++
 xattr_unsupported.go | 3 +++
 4 files changed, 12 insertions(+)

diff --git a/xattr_bsd.go b/xattr_bsd.go
index 09db849..183730a 100644
```

I also seem to get a few extra lines. Don't we need to set level to match the preamble length?

Reply:
Patches written by `git format-patch` are in mbox format, and that specific date in the first line is used by Git to indicate that the patch was written by `format-patch`.

`-p` strips leading components from the paths of files to be patched, so `-p1` is the correct level for `git format-patch`-formatted patches (the leading `a/` and `b/` need to be stripped).